File: atcoder/abc/abc156/e/main.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long 

int main(){
  int NN = 3e5;
  ll mod = 1e9+7;
  vector<ll> fac(NN, 0), finv(NN, 0), inv(NN, 0);

  fac[0] = fac[1] = 1;
  finv[0] = finv[1] = 1;
  inv[1] = 1;

  for(int i=2;i<NN;i++){
    fac[i] = fac[i-1]*i%mod;
    inv[i] = mod-inv[mod%i]*(mod/i)%mod;
    finv[i] = finv[i-1]*inv[i]%mod;
  }

  ll N, K;
  cin >> N >> K;
  ll ans = 0;
  for(int i=0;i<min(N, K+1);i++){
    //ans += nCi * N-iHi
    ans += fac[N]%mod*finv[i]%mod*finv[N-i]%mod*fac[N-1]%mod*finv[i]%mod*finv[N-i-1]%mod;
    ans %= mod;
  }
  cout << ans << endl;
}

File: atcoder/abc/abc156/e/main.py
mod = 10**9+7
NN = 3*10**5
fac = [0]*NN
finv = [0]*NN
inv = [0]*NN
fac[0] = fac[1] = 1
finv[0] = finv[1] = 1
inv[1] = 1
for i in range(2, NN):
    fac[i] = fac[i-1]*i%mod
    inv[i] = mod-inv[mod%i]*(mod//i)%mod
    finv[i] = finv[i-1]*inv[i]%mod

def comb(a, b, p=10**9+7):
    if a<b: return 0
    elif a<0 or b<0: return 0
    return fac[a]*(finv[b]*finv[a-b]%p)%p



N, K = map(int, input().split())
ans = 0
for i in range(min(N, K+1)):
    ans += comb(N, i)%mod*comb(N-1,i)%mod
    ans %= mod
    #print(i, ans)
print(ans)